lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -3055,15 +3042,28 @@ class VJPEmitter final
 
   /// Build a predecessor enum instance using the given builder for the given
   /// original predecessor/successor blocks and pullback struct value.
-  EnumInst *buildPredecessorEnumValue(
-      SILBuilder &builder, SILBasicBlock *predBB, SILBasicBlock *succBB,
-      StructInst *pbStructVal) {
+  EnumInst *buildPredecessorEnumValue(SILBuilder &builder,
+                                      SILBasicBlock *predBB,
+                                      SILBasicBlock *succBB,
+                                      StructInst *pbStructVal) {
     auto loc = pbStructVal->getLoc();
     auto *succEnum = pullbackInfo.getPredecessorEnum(succBB);
     auto enumLoweredTy = getNominalDeclLoweredType(succEnum);
     auto *enumEltDecl =
         pullbackInfo.lookUpPredecessorEnumElement(predBB, succBB);
-    return builder.createEnum(loc, pbStructVal, enumEltDecl, enumLoweredTy);
+    auto enumEltType = remapType(
+        enumLoweredTy.getEnumElementType(enumEltDecl, getModule()));
+    // If the enum element type does not have a box type (i.e. the enum case is
+    // not indirect), then directly create an enum.
+    auto boxType = dyn_cast<SILBoxType>(enumEltType.getASTType());
+    if (!boxType)
+      return builder.createEnum(loc, pbStructVal, enumEltDecl, enumLoweredTy);
+    // Otherwise, box the pullback struct value and create an enum.
+    auto *allocBox = builder.createAllocBox(loc, boxType);
```

Comment:
This `alloc_box` is not being deallocated and may cause leaks - to be verified.
Loop AD leak checking tests are added in `test/AutoDiff/leakchecking.swift`.

Reply:
Resolved - see [this comment](https://github.com/apple/swift/pull/25558#pullrequestreview-250897375). Deallocation is handled automatically during `release_value` for pullback structs.